perun-core/src/test/java/cz/metacentrum/perun/core/entry/UsersManagerEntryIntegrationTest.java

```diff
@@ -393,32 +392,155 @@ public void updateUserWithNullValueInLastName() throws Exception {
+		attrLogin2.setType(String.class.getName());
+		attrLogin2 = new Attribute(perun.getAttributesManager().createAttribute(sess, attrLogin2));
+		attrLogin2.setValue("login2");
+		perun.getAttributesManager().setAttribute(sess, user, attrLogin2);
+
+		boolean originalUserDeletionForced = BeansUtils.getCoreConfig().getUserDeletionForced();
+		try {
+			// Enable deletion of users
+			BeansUtils.getCoreConfig().setUserDeletionForced(true);
+			usersManager.deleteUser(sess, user, true);
+		} finally {
+			// set userDeletionForced back to the original value
+			BeansUtils.getCoreConfig().setUserDeletionForced(originalUserDeletionForced);
+		}
+
+		assertTrue(usersManager.isLoginBlockedForNamespace(sess, "login1", "namespace1", true));
+		assertFalse(usersManager.isLoginBlockedForNamespace(sess, "login1", "namespace2", true));
+		assertTrue(usersManager.isLoginBlockedForNamespace(sess, "login2", "namespace2", true));
+		assertFalse(usersManager.isLoginBlockedForNamespace(sess, "login2", "namespace1", true));
+
+		assertEquals("The number of blocked logins should be 2.", 2, usersManager.getAllBlockedLoginsInNamespaces(sess).size());
+
+		assertThatExceptionOfType(UserNotExistsException.class).isThrownBy(
+			() -> perun.getUsersManager().getUserById(sess, user.getId()));
+	}
+
+	@Test
+	public void blockedLoginsAndRelatedUserIds() throws Exception {
+		System.out.println(CLASS_NAME + "blockedLoginsAndRelatedUserIds");
+
+		String login = "login1";
+		String namespace = "namespace1";
+
+		perun.getUsersManager().blockLogins(sess, Collections.singletonList(login), namespace);
+
+		// user id should NOT be stored with block login - method should return null
+		assertNull(usersManager.getRelatedUserIdByBlockedLoginInNamespace(sess, login, namespace));
+
+		// Create login for user in namespace1
+		Attribute attrLogin = new Attribute();
+		attrLogin.setNamespace(AttributesManager.NS_USER_ATTR_DEF);
+		attrLogin.setFriendlyName("login-namespace:namespace2");
+		attrLogin.setType(String.class.getName());
+		attrLogin = new Attribute(perun.getAttributesManager().createAttribute(sess, attrLogin));
+		attrLogin.setValue("login2");
+		perun.getAttributesManager().setAttribute(sess, user, attrLogin);
+
+		boolean originalUserDeletionForced = BeansUtils.getCoreConfig().getUserDeletionForced();
+		try {
+			// Enable deletion of users
+			BeansUtils.getCoreConfig().setUserDeletionForced(true);
+			usersManager.deleteUser(sess, user, true);
+		} finally {
+			// set userDeletionForced back to the original value
+			BeansUtils.getCoreConfig().setUserDeletionForced(originalUserDeletionForced);
+		}
+
+		// check that user id is correctly stored with the block login
+		assertEquals(user.getId(), usersManager.getRelatedUserIdByBlockedLoginInNamespace(sess, "login2", "namespace2").intValue());
+		assertThatExceptionOfType(LoginIsNotBlockedException.class).isThrownBy(
+			() -> usersManager.getRelatedUserIdByBlockedLoginInNamespace(sess, "login3", "namespace2"));
+	}
```

Comment:
Not sure what is our convention but right now the test tests for two different things ? That login was correctly blocked with user id and that if some login is not blocked exception is thrown. If you see this as a problem please create a separate test for one of the cases.

Reply:
We discussed it with @xflord and we agreed that there it is probably not necessary - these asserts are both related to the same method and this method even is not in openapi, so I consider it just as a side assert.